src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,33 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final String LOGIN_ALREADY_TAKEN = "This login is already taken";
+    private static final String LOGIN_TOO_SHORT = "Login cannot be null or less than 6 characters";
+    private static final String PASSWORD_TOO_SHORT
+            = "Password cannot be null or less than 6 characters";
+    private static final String AGE_TOO_YOUNG = "User's age must be at least 18 and cannot be null";
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException(LOGIN_TOO_SHORT);
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException(PASSWORD_TOO_SHORT);
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
```

Comment:
The condition `user.getAge() == null` should be checked separately before comparing the age to 18. The current condition might throw a `NullPointerException` if `user.getAge()` is null.